core/variant/binder_common.h

```diff
@@ -82,112 +82,10 @@ struct VariantCaster<const T &> {
-			return BitField<m_enum>(*reinterpret_cast<const int64_t *>(p_ptr));        \
-		}                                                                              \
-		typedef int64_t EncodeT;                                                       \
-		_FORCE_INLINE_ static void encode(BitField<m_enum> p_val, const void *p_ptr) { \
-			*(int64_t *)p_ptr = p_val;                                                 \
-		}                                                                              \
-	};                                                                                 \
-	template <>                                                                        \
-	struct ZeroInitializer<BitField<m_enum>> {                                         \
-		static void initialize(BitField<m_enum> &value) {                              \
-			value = 0;                                                                 \
-		}                                                                              \
-	};                                                                                 \
-	template <>                                                                        \
-	struct VariantInternalAccessor<BitField<m_enum>> {                                 \
-		static _FORCE_INLINE_ BitField<m_enum> get(const Variant *v) {                 \
-			return BitField<m_enum>(*VariantInternal::get_int(v));                     \
-		}                                                                              \
-		static _FORCE_INLINE_ void set(Variant *v, BitField<m_enum> p_value) {         \
-			*VariantInternal::get_int(v) = p_value.operator int64_t();                 \
-		}                                                                              \
-	};
-
-// Object enum casts must go here
-VARIANT_ENUM_CAST(Object::ConnectFlags);
-
-VARIANT_ENUM_CAST(Vector2::Axis);
-VARIANT_ENUM_CAST(Vector2i::Axis);
-VARIANT_ENUM_CAST(Vector3::Axis);
-VARIANT_ENUM_CAST(Vector3i::Axis);
-VARIANT_ENUM_CAST(Vector4::Axis);
-VARIANT_ENUM_CAST(Vector4i::Axis);
-VARIANT_ENUM_CAST(EulerOrder);
-VARIANT_ENUM_CAST(Projection::Planes);
-
-VARIANT_ENUM_CAST(Error);
-VARIANT_ENUM_CAST(Side);
-VARIANT_ENUM_CAST(ClockDirection);
-VARIANT_ENUM_CAST(Corner);
-VARIANT_ENUM_CAST(HatDir);
-VARIANT_BITFIELD_CAST(HatMask);
-VARIANT_ENUM_CAST(JoyAxis);
-VARIANT_ENUM_CAST(JoyButton);
-
-VARIANT_ENUM_CAST(MIDIMessage);
-VARIANT_ENUM_CAST(MouseButton);
-VARIANT_BITFIELD_CAST(MouseButtonMask);
-VARIANT_ENUM_CAST(Orientation);
-VARIANT_ENUM_CAST(HorizontalAlignment);
-VARIANT_ENUM_CAST(VerticalAlignment);
-VARIANT_ENUM_CAST(InlineAlignment);
-VARIANT_ENUM_CAST(PropertyHint);
-VARIANT_BITFIELD_CAST(PropertyUsageFlags);
-VARIANT_ENUM_CAST(Variant::Type);
-VARIANT_ENUM_CAST(Variant::Operator);
-
-// Key
-
-VARIANT_ENUM_CAST(Key);
-VARIANT_BITFIELD_CAST(KeyModifierMask);
-VARIANT_ENUM_CAST(KeyLocation);
+#ifndef DISABLE_DEPRECATED
+#define VARIANT_ENUM_CAST(m_enum) // Does nothing.
```

Comment:
Are these stubs still needed? If they are not used in the codebase (you seem to have removed a lot of them), maybe an [explicit deprecation](https://stackoverflow.com/a/29297970/730797) would be good?

Reply:
I suppose not. I kept them for compatibility with third-party modules, but their removal isn't technically "breaking" unless you do a deliberate rescope like `RenderingDevice`. Will likely remove next rebase.